### ISSUE TITLE
FR: Please support human-friendly project names

### ISSUE THREAD

**Issue by bukzor-sentryio** [google-github-actions/auth#341]
### TL;DR

Currently the action will fail if given `/project/projectname/...` as the `workload_identity_provider`. The error indicates the failure is due to a mismatch in the `audience` field of the resulting request.

### Detailed design

If the project is given in non-numeric form, make an additional request to normalize its form.

A more exact description in python psuedocode:

```python
POOL_ID = "/project/my-project/locations/..."
def norm_numeric_project(id):
  parts = id.split("/")
  assert parts[:2] == ['', 'projects'], parts
  project = parts[2]
  if not project.isnumeric():
    project = subprocess.check_output(('gcloud', 'projects', 'describe', project, '--format', 'get(projectNumber)'))
  parts[1] = project
  return '/'.join(parts)
```

### Additional information

Supporting the human friendly (i.e. `/project/$GOOGLE_PROJECT/...`) ID will make our usages of this action considerably more clear -- which is important for security. It's also less-surprising behavior for users.

**Comment by sethvargo** [comment]
Hi @bukzor-sentryio - this is a limitation of the Workload Identity product, it requires using a project number and does not support using a project id. Workload Identity is a feature of IAM. You can open an issue with the IAM team by creating a public issue on their issuer tracker: https://b.corp.google.com/savedsearches/559761
